internal/repositories/devicedefinition/devicedefinition.go

```diff
@@ -0,0 +1,239 @@
+	if err != nil {
+		return nil, gqlerror.Errorf("failed load manufactures: %s", err)
+	}
+
+	var manufacturer *services.ManufacturerCacheModel
+	for _, model := range manufactures {
+		if model.Slug == manufacturerSlug {
+			manufacturer = &model
+			break
+		}
+	}
+
+	if manufacturer == nil {
+		return nil, gqlerror.Errorf("Manufacturer %s doesn't exist.", manufacturerSlug)
+	}
+
+	tableName, err := r.ManufacturerContractService.GetTableName(ctx, manufacturer.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	statement := fmt.Sprintf("SELECT * FROM %s WHERE id = '%s'", *tableName, by.ID)
+	queryParams := map[string]string{
+		"statement": statement,
+	}
+	var modelTableland []DeviceDefinitionTablelandModel
+
+	if err = r.TablelandApiService.Query(ctx, queryParams, &modelTableland); err != nil {
+		return nil, err
+	}
+
+	for _, item := range modelTableland {
+		return ToAPI(manufacturerSlug, &item), nil
+	}
+
+	return nil, nil
+}
+
+func (r *Repository) GetDeviceDefinitions(ctx context.Context, first *int, after *string, last *int, before *string, filterBy *gmodel.DevicedefinitionFilter) (*gmodel.DeviceDefinitionConnection, error) {
+	limit, err := helpers.ValidateFirstLast(first, last, base.MaxPageSize)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(filterBy.Manufacturer) == 0 {
+		return nil, gqlerror.Errorf("Provide exactly one `manufacturer`.")
+	}
+
+	var conditions []string
+	if filterBy.Year != nil && (*filterBy.Year > 1980 && *filterBy.Year < 2999) {
+		conditions = append(conditions, fmt.Sprintf("year = %d", *filterBy.Year))
+	}
+	if filterBy.Model != nil && len(*filterBy.Model) > 0 {
+		conditions = append(conditions, fmt.Sprintf("model = '%s'", *filterBy.Model))
+	}
+	if filterBy.ID != nil && len(*filterBy.ID) > 0 {
+		conditions = append(conditions, fmt.Sprintf("id = '%s'", *filterBy.ID))
+	}
+
+	whereClause := strings.Join(conditions, " AND ")
+	if whereClause != "" {
+		whereClause = " WHERE " + whereClause
+	}
+
+	manufactures, err := r.ManufacturerCacheService.GetAllManufacturers(ctx)
+	if err != nil {
+		return nil, gqlerror.Errorf("failed load manufactures: %s", err)
+	}
+
+	var manufacturer *services.ManufacturerCacheModel
+	for _, model := range manufactures {
+		if model.Slug == filterBy.Manufacturer {
+			manufacturer = &model
+			break
+		}
+	}
+
+	tableName, err := r.ManufacturerContractService.GetTableName(ctx, manufacturer.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	countParams := map[string]string{
+		"statement": fmt.Sprintf("SELECT count(*) FROM %s%s", *tableName, whereClause),
+	}
+	var modelCountTableland []DeviceDefinitionTablelandCountModel
+	if err = r.TablelandApiService.Query(ctx, countParams, &modelCountTableland); err != nil {
+		return nil, err
+	}
+
+	totalCount := modelCountTableland[0].Count
+
+	queryParams := map[string]string{
+		"statement": fmt.Sprintf("SELECT * FROM %s%s LIMIT %d OFFSET %d", *tableName, whereClause, limit, 1),
```

Comment:
Are we ordering this in any way? It doesn't feel like pagination is going to work. Why offset 1?